app/src/main/java/com/navapp/navigation/requests/RoutePlanner.java

```diff
@@ -0,0 +1,45 @@
+package com.navapp.navigation.requests;
+
+import com.google.maps.DirectionsApi;
+import com.google.maps.GeoApiContext;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.GeocodedWaypoint;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RoutePlanner {
+    private final GeoApiContext context;
+
+    public RoutePlanner(GeoApiContext context) {
+        this.context = context;
+    }
+
+    public List<String> orderedRouteIds(String originId, List<String> waypointIds, String destinationId) throws InterruptedException, ApiException, IOException {
+        GeocodedWaypoint[] orderedWaypoints = DirectionsApi.newRequest(context)
+                .originPlaceId(originId)
+                .waypointsFromPlaceIds(toArray(waypointIds))
+                .destinationPlaceId(destinationId)
+                .optimizeWaypoints(true)
+                .await().geocodedWaypoints;
```

Comment:
Do you understand the implications of waiting here?

Reply:
Do you understand the implications of not?

Reply:
Hmmmmmmmmmm
`await` is blocking. Make note of that, because depending on how it used, this may be a problem.

Reply:
Of course, but I don't really have a choice not to wait.... These exceptions are thrown by it, so it's gonna be easy to handle a timeout or something

Reply:
Moved to the `PlaceAutocomplete` because it's the same conversation.